### PR TITLE
[DOC] Update contributing guide project section

### PR DESCRIPTION
**Pull request by PaulRMellor**
Signed-off-by: prmellor <pmellor@redhat.com>

**Documentation**

Update to the _Project layout_ section of the Contributing Guide.
Added the new _Deploying_ guide, plus clarified what the doc folder contains and what each guide is intended for.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

**Comment by PaulRMellor** [comment]
Not sure if we need to keep this section:

https://strimzi.io/contributing/guide/#dependencies

If we do, then maybe we should explain a bit more.

File: documentation/contributing/introduction.adoc [hunk 1]
```diff
@@ -9,13 +9,52 @@ This guide is meant to assist contributors to the Strimzi documentation.
 
 == Project layout
 
+Strimzi uses git to manage repositories.
+
+https://github.com/strimzi/strimzi-kafka-operator[`strimzi-kafka-operator`^] (GitHub):: The public GitHub repo hosts all of the Strimzi code and documentation.
+https://github.com/strimzi/strimzi-kafka-operator/tree/master/documentation[`strimzi-kafka-operator/documentation`^] (GitHub):: The _documentation_ folder is split into _category_ folders to manage the content.
+
+Category folders contain files related to Strimzi guides (Deploying, Quickstart, Overview, Using), and the files that provide the content for one or more of these guides – _assemblies_ and _modules_.
+Assemblies, which usually encapsulate a feature or process, bring the related content contained in modules together.
```
Comment (laidan6000) [inline documentation/contributing/introduction.adoc:18]: An analogy would be helpful here. 

An assembly is like a sub-section or chapter in a printed book. For example, a book on _Event-driven architectures_ might have a chapter on _Stream processing_, which would be an assembly.

File: documentation/contributing/introduction.adoc [hunk 2]
```diff
@@ -9,13 +9,52 @@ This guide is meant to assist contributors to the Strimzi documentation.
...
+=== Strimzi Overview guide
+
+The intention of the _Strimzi Overview guide_ is for developing an understanding of Strimzi and Apache Kafka.
+The guide does not contain any instructions.
+It provides an overview of the concepts behind Apache Kafka, the Kafka component architecture, and how Strimzi supports Kafka in a Kubernetes cluster.
+The guide also describes how Strimzi _Operators_ help manage a deployment.
+
+The guide contains high-level outlines of the processes required to deploy, configure, secure and monitor a deployment of Strimzi.
+
+=== Deploying Strimzi
```
Comment (laidan6000) [inline documentation/contributing/introduction.adoc:42]: Inconsistency in the title.

File: documentation/contributing/introduction.adoc [hunk 3]
```diff
@@ -9,13 +9,52 @@ This guide is meant to assist contributors to the Strimzi documentation.
...
+
+The _Deploying Strimzi_ guide provides instructions on all the options available for deploying and upgrading Strimzi.
+The guide describes what is deployed, and the order of deployment required.
+
+As well as describing the deployment steps, the guide also provides pre- and post-deployment instructions to prepare for and verify a deployment.
+Additional deployment options described include the steps to introduce metrics.
+
```
Comment (laidan6000) [inline documentation/contributing/introduction.adoc:49]: Also mention upgrading.

File: documentation/contributing/introduction.adoc [hunk 4]
```diff
@@ -9,13 +9,52 @@ This guide is meant to assist contributors to the Strimzi documentation.
...
+=== Strimzi Quick Start guide
+
```
Comment (laidan6000) [inline documentation/contributing/introduction.adoc:51]: Could state that the Quick Start uses MiniKube.

File: documentation/contributing/introduction.adoc [hunk 5]
```diff
@@ -9,13 +9,52 @@ This guide is meant to assist contributors to the Strimzi documentation.
...
+The _Strimzi Quick Start guide_ provides instructions for evaluating Strimzi.
+The steps describe how to get a Strimzi deployment up-and-running as quickly as possible.
+
+In contrast to the _Deploying_ guide, this guide provides a reduced set of instructions for a specific type of deployment, with minimal configuration.
+
+=== Using Strimzi
+
+The _Using Strimzi_ guide provides configuration instructions for Kafka components, and instructions for using Strimzi Operators.
+The majority of the content relates to how you might want to modify your deployment and introduce additional features, such as cruise control or distributed tracing.
```
Comment (laidan6000) [inline documentation/contributing/introduction.adoc:60]: Cruise Control

**Comment by laidan6000** [comment]
> Not sure if we need to keep this section:
> 
> https://strimzi.io/contributing/guide/#dependencies
> 
> If we do, then maybe we should explain a bit more.

I would keep it because a contributor needs asciidoctor to build and preview their updates.

I think we should mention the `make` targets here, which are optional.

(Optional) The `yq` YAML processor: https://mikefarah.gitbook.io/yq/.

You only need `yq` to build the documentation using the `make` targets. It is simpler to use Asciidoctor to preview most documentation updates.

File: documentation/contributing/introduction.adoc [hunk 6]
```diff
@@ -9,13 +9,52 @@ This guide is meant to assist contributors to the Strimzi documentation.
...
+https://github.com/strimzi/strimzi-kafka-operator[`strimzi-kafka-operator`^] (GitHub):: The public GitHub repo hosts all of the Strimzi code and documentation.
+https://github.com/strimzi/strimzi-kafka-operator/tree/master/documentation[`strimzi-kafka-operator/documentation`^] (GitHub):: The _documentation_ folder is split into _category_ folders to manage the content.
+
+Category folders contain files related to Strimzi guides (Deploying, Quickstart, Overview, Using), and the files that provide the content for one or more of these guides – _assemblies_ and _modules_.
+Assemblies, which usually encapsulate a feature or process, bring the related content contained in modules together.
+
 [source,options="nowrap",subs="+quotes"]
 ----
-quickstart/                          # Quickstart guide
+deploying/                           # Deploying and Upgrading guide
```
[the remark this answers is not in the document]
Comment (PaulRMellor) [reply documentation/contributing/introduction.adoc:22]: I've changed all to the book titles here

File: documentation/contributing/introduction.adoc [hunk 7]
```diff
@@ -9,13 +9,52 @@ This guide is meant to assist contributors to the Strimzi documentation.
 
 == Project layout
 
+Strimzi uses git to manage repositories.
+
+https://github.com/strimzi/strimzi-kafka-operator[`strimzi-kafka-operator`^] (GitHub):: The public GitHub repo hosts all of the Strimzi code and documentation.
```
[the remark this answers is not in the document]
Comment (PaulRMellor) [reply documentation/contributing/introduction.adoc:14]: I've added a link to the kafka Bridge repo and mentioned Asciidoc in the introduction.

File: documentation/contributing/introduction.adoc (hunk 1 above)
Comment (PaulRMellor) [reply documentation/contributing/introduction.adoc:18]: An assembly is like a sub-section or chapter in a book.
For example, the _Deploying Strimzi_ guide has a chapter called _Verifying the {ProductName} deployment_, which is contained in its own assembly.

File: documentation/contributing/introduction.adoc (hunk 2 above)
Comment (PaulRMellor) [reply documentation/contributing/introduction.adoc:42]: That's the name at the moment. I've reflected it in the list too.

File: documentation/contributing/introduction.adoc (hunk 3 above)
Comment (PaulRMellor) [reply documentation/contributing/introduction.adoc:49]: Instructions are provided for Strimzi and Kafka upgrades.

File: documentation/contributing/introduction.adoc (hunk 4 above)
Comment (PaulRMellor) [reply documentation/contributing/introduction.adoc:51]: Good idea.

**Comment by PaulRMellor** [comment]
> > Not sure if we need to keep this section:
> > https://strimzi.io/contributing/guide/#dependencies
> > If we do, then maybe we should explain a bit more.
> 
> I would keep it because a contributor needs asciidoctor to build and preview their updates.
> 
> I think we should mention the `make` targets here, which are optional.
> 
> * (Optional) The `yq` YAML processor: https://mikefarah.gitbook.io/yq/
> 
> You only need `yq` to build the documentation using the `make` targets. For most documentation updates, it is simpler to use Asciidoctor.

Thanks Daniel. I've update the section as follows:

image

**Comment by scholzj** [comment]
@ppatierno could you please review this?

File: documentation/contributing/introduction.adoc [hunk 8]
```diff
@@ -1,21 +1,71 @@
...
+https://github.com/mikefarah/yq[`yq`^]:: YAML build tool to build documentation using _make_ targets
+
+For most documentation updates, `asciidoctor` offers the simplest way to check the build.
 
 == Project layout
 
+Strimzi uses git to manage repositories.
+
+https://github.com/strimzi/strimzi-kafka-operator[`strimzi-kafka-operator`^] (GitHub):: The public GitHub repo hosts all of the Strimzi code and documentation.
+https://github.com/strimzi/strimzi-kafka-bridge[`strimzi-kafka-bridge`^] (GitHub):: The public GitHub repo hosts all of the Strimzi Kafka Bridge code.
```
Comment (ppatierno) [inline documentation/contributing/introduction.adoc:20]: We have the bridge here, should we add the `oauth` repo as well?

Comment (scholzj) [reply documentation/contributing/introduction.adoc:20]: @PaulRMellor Can you have a look at this? It would be great if we could close this PR finally.

**Pull request merged**